crypto/getenv.c

```diff
@@ -13,9 +13,45 @@
 
 #include <stdlib.h>
 #include "internal/cryptlib.h"
+#include "e_os.h"
 
 char *ossl_safe_getenv(const char *name)
 {
+#if (defined(OPENSSL_SYS_WINDOWS)) && defined(CP_UTF8) && !defined(_WIN32_WCE)
+
+    if (GetEnvironmentVariableW(L"OPENSSL_WIN32_UTF8", NULL, 0) == 0)
+        return getenv(name);
+
+    /* convert name to wide string */
+    int rsize = mbstowcs(NULL, name, 0) + 1;
+    WCHAR * namew = _alloca(rsize * sizeof(WCHAR));
+    int fsize = mbstowcs(namew, name, rsize);
+
+    char *val = NULL;
+
+    /* determine value string size in wchars */
+    DWORD envlen = GetEnvironmentVariableW(namew, NULL, 0);    
+
+    if (envlen != 0) {
+        WCHAR *valw = _alloca(envlen * sizeof(WCHAR));
+        if (GetEnvironmentVariableW(namew, valw, envlen) < envlen) {
```

Comment:
The last param to GetEnvironmentVariableW includes the null terminator.

Reply:
MS DOC:

If the function succeeds, the return value is the number of characters stored in the buffer pointed to by lpBuffer, not including the terminating null character.

Reply:
https://docs.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-getenvironmentvariablew

Reply:
working example:

env TEST=12345

GetEnvironmentVariableW('TEST', NULL, 0) => 6

(GetEnvironmentVariableW('TEST', valw, 6) => 5

where is problem?